### PR TITLE
swarm/api: manifests cleanups

### PR DESCRIPTION
**Pull request by acud**

### CODE REVIEW

File: swarm/api/manifest.go [hunk 1]
```diff
@@ -578,9 +590,284 @@ func RegularSlashes(path string) (res string) {
...
 func (mt *manifestTrie) getEntry(spath string) (entry *manifestTrieEntry, fullpath string) {
+	log.Debug("api.manifest.getEntry", "spath", spath)
 	path := RegularSlashes(spath)
 	var pos int
 	quitC := make(chan bool)
 	entry, pos = mt.findPrefixOf(path, quitC)
 	return entry, path[:pos]
 }
+
+func (a *API) UpdateManifest(ctx context.Context, addr storage.Address, update func(mw *ManifestWriter) error) (storage.Address, error) {
```
Comment (zelig) [inline swarm/api/manifest.go:600]: why are these moved?

Comment (acud) [reply swarm/api/manifest.go:600]: ideally we should have 1 type `Manifest` that you can perform an `Update` action on. right now the code that does manifest modifications is spread out on different files, contexts and types. it should all be governed from one place.

**Comment by acud** [comment]
closing this as better design was reached and will reiterate over this topic once bandwidth permits.
